### DISCUSSION TITLE
How to deal with rebadged/OEM hardware?

### DISCUSSION THREAD

**Discussion by mgoetze5** [netbox-community/netbox#5766] [Q&A] [answered]
I have a bunch of Fibre Channel switches such as

Brocade 6510
HPE SN6000B
EMC DS-6510B

which are exactly the same hardware, just with a different manufacturer logo glued to the front. Is there a way to represent this in NetBox? As far as I understand, I can only choose a platform from the manufacturer of the device, so if I were to treat them all as completely different, I would have to add "FabricOS 8.2.2a" etc. as a platform for Brocade, HPE and EMC, is that correct?

**Comment by candlerb** [comment, marked as answer]
In Netbox, platforms can either be generic (linked to no manufacturer, and usable on any manufacturer's device), or tied to a single manufacturer.

If you don't want "FabricOS" to be available as an option across all your servers, but only on those three manufacturers, then yes you'd need to add it three times.

**Reply by mgoetze5** [reply]
OK, and there's also no way to link device types? I tried creating the switch for one manufacturer and then cloning it, and the clone didn't have any of the interfaces, is that working as designed?

**Reply by candlerb** [reply]
> OK, and there's also no way to link device types?

That's correct.  In addition, you can have different Cisco devices which run different types of platform; but they are all available to any device with manufacturer "Cisco" (so it lets you select IOS for your Nexus switch)

> I tried creating the switch for one manufacturer and then cloning it, and the clone didn't have any of the interfaces, is that working as designed?

As far as I can see, "clone device" is simply "create new device (of the same type and role)".  The interfaces to create are picked up from the device type template, not the device being cloned.

I guess cloning the actual device interfaces might be more useful - but that would be a feature request for a behaviour change.
